src/Microsoft.DncEng.Configuration.Extensions/KeyVaultConfigMapper.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Text.RegularExpressions;
+using Azure;
+using Azure.Security.KeyVault.Secrets;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.DncEng.Configuration.Extensions
+{
+    public static class KeyVaultConfigMapper
+    {
+        private static readonly Regex VaultReferenceRegex = new Regex(@"\[vault\((?<key>[^])]+)\)\]");
+
+        public static Func<string, string> Create(IConfiguration bootstrapConfiguration)
```

Comment:
This is identical to the previous class except the regex and line 25... that seems like we could factor stuff a bit.

Reply:
I'll factor some stuff out.